includes/class-sensei-quiz.php

```diff
@@ -1523,6 +1528,53 @@ public static function start_quiz_questions_loop() {
 		$sensei_question_loop['quiz_id']   = $quiz_id;
 	}
 
+	/**
+	 * Display quiz progress bar.
+	 *
+	 * @param int   $quiz_id quiz id.
+	 * @param array $pagination_settings pagination settings.
+	 * @since 3.15.0
+	 */
+	public static function display_progress_bar( $quiz_id, $pagination_settings ) {
```

Comment:
Are there any issues with merging this method into the `the_quiz_progress_bar` method? I think this might simplify things.

My idea is for the method to do the following: get/check the progress bar settings, enqueue the needed scripts, render the template.

Reply:
So that was the original idea, but the issue there is that I don't have jet all the data that I need. That method is being called on ```sensei_single_quiz_questions_before``` so that the template gets rendered before the quiz.

Reply:
You can get the data from the `$sensei_question_loop` global make a query for the pagination settings (meta queries are cached if called multiple times).

Here's a quick example (which I haven't tested that much) of the combined methods:

```php
	/**
	 * Display the quiz progress bar when the setting is enabled and the quiz is paginated.
	 *
	 * @since 3.15.0
	 */
	public static function the_quiz_progress_bar() {
		global $sensei_question_loop;

		$quiz_id             = get_the_ID();
		$pagination_settings = json_decode(
			get_post_meta( $quiz_id, '_pagination', true ),
			true
		);

		// Make sure the quiz is paginated and the progress bar enabled.
		if ( $sensei_question_loop['total_pages'] <= 1 || empty( $pagination_settings['show_progress_bar'] ) ) {
			return;
		}

		$user_id       = get_current_user_id();
		$lesson_id     = Sensei()->quiz->get_lesson_id( $quiz_id );
		$answers       = Sensei()->quiz->get_user_answers( $lesson_id, $user_id );
		$answers_count = is_array( $answers ) ? count( array_filter( $answers ) ) : 0;

		Sensei()->assets->enqueue( 'sensei-shared-blocks-style', 'blocks/shared-style.scss' );
		wp_enqueue_script( 'sensei-quiz-progress' );
		wp_localize_script(
			'sensei-quiz-progress',
			'progress_bar_properties',
			array(
				'totalNumber'     => $sensei_question_loop['total'],
				'completedNumber' => $answers_count,
				'radius'          => $pagination_settings['progress_bar_radius'],
				'height'          => $pagination_settings['progress_bar_height'],
				'color'           => empty( $pagination_settings['progress_bar_color'] ) ? '' : $pagination_settings['progress_bar_color'],
				'backgroundColor' => empty( $pagination_settings['progress_bar_background'] ) ? '' : $pagination_settings['progress_bar_background'],
			)
		);

		Sensei_Templates::get_template( 'globals/progress-bar.php' );
	}
```

Reply:
So the things is that ```sensei_question_loop``` is empty at that moment

Reply:
https://github.com/Automattic/sensei/pull/4606/files#diff-964a2368a54a4fd7dd7d8dfe593a8f74e690d028590dc23dbb8d1d4e5b68577fR1643